ebu_tt_live/bindings/converters/ebutt3_ebuttd.py

```diff
@@ -163,35 +163,39 @@ def _fix_fontsize(self, elem, celem, parent, dataset):
 
             computed_font_size = elem.computed_style.fontSize
             computed_line_height = elem.computed_style.lineHeight
+            parent_computed_font_size = DEFAULT_CELL_FONT_SIZE
+            if isinstance(parent, (body_type, div_type, p_type, span_type)):
+                if parent.computed_style.fontSize is not None:
+                    parent_computed_font_size = parent.computed_style.fontSize
 
             required_font_size = None
             required_line_height = None
 
             if specified_font_size is not None:
                 # Fallback for body element fontSize is the default value
                 # because EBU-TT Live does not allow fontSize on region
-                # elements.
-                if isinstance(elem, body_type) and computed_font_size is None:
+                # elements. Note the assumption written above is probably
+                # wrong: a region can reference a style with a fontSize
+                # element. This is somewhat confusing in the specification.
+                if isinstance(elem, (body_type, region_type)) and computed_font_size is None:
                     computed_font_size = DEFAULT_CELL_FONT_SIZE
 
-                if isinstance(elem, (body_type, div_type, p_type)):
+                if isinstance(elem, (region_type, body_type, div_type, p_type, span_type)):
                     # Since we eliminated all our fontSize attributes from the
                     # original styles here it is as simple as computing based
                     # on the default value.
                     if isinstance(computed_font_size,
                                   CellFontSizeType) \
                        and computed_font_size != DEFAULT_CELL_FONT_SIZE:
-                        required_font_size = \
-                            computed_font_size / DEFAULT_CELL_FONT_SIZE
+
+                        if parent_computed_font_size.vertical != computed_font_size.vertical:
+                            required_font_size = \
+                                computed_font_size / parent_computed_font_size
+
                     elif isinstance(computed_font_size,
                                     PercentageFontSizeType):
-                        required_font_size = computed_font_size
-
-                elif isinstance(elem, span_type):
-                    parent_computed_font_size = parent.computed_style.fontSize
-                    if parent_computed_font_size != computed_font_size:
-                        required_font_size = \
-                            computed_font_size / parent_computed_font_size
+                        if computed_font_size.vertical != 100:
+                            required_font_size = computed_font_size
 
             if specified_line_height is not None:
```

Comment:
We probably should be doing something tidier with lineHeight as well. I just ran a test file through and found a div getting a style with fontSize 150% and lineHeight 120%, and one of its child p elements had a style with lineHeight 120%. This latter is pointless, and makes no difference, since lineHeight is inherited and percentages relate to the computed font size, not the inherited computed font size. So an improvement (perhaps done separately) would be to strip out lineHeight styles that only set the lineHeight to the inherited value.

Reply:
This is done in 2dc3b91.